utility/hardware_bom/hardware_bom_purchases.py

```diff
@@ -0,0 +1,74 @@
+
+import pandas as pd
+
+# parse_hardware_bom will accept the BOM.csv and MS_Quote Excel File and return two seperate Excel/CSV files with winning parts. 
+def parse_hardware_bom(bom_file, ms_quote_file): 
+
+    bom_df = pd.read_csv(bom_file)
+    ms_quote_df = pd.read_excel(ms_quote_file)
+    
+    #  Create DF of relevant information.
+    bom_df = bom_df[['Manufacturer Part Number', 'Description', 'Unit Price', 'Mfg Std Lead Time']]
+    ms_quote_df = ms_quote_df[['Mfr. #', 'Description', 'Order Unit Price', 'Lead Time in Weeks', 'Priority']] 
+    bom_df = bom_df.rename(columns = {'Manufacturer Part Number':'Mfr. #'}) 
+
+    # Fill Nan values with zeroes for calculation purposes.
+    bom_df['Mfg Std Lead Time'] = bom_df['Mfg Std Lead Time'].fillna(0) 
+    ms_quote_df['Lead Time in Weeks'] = ms_quote_df['Lead Time in Weeks'].fillna(0)
+
+    # Correct issue of having "$" in front of numeric price value.
+    ms_quote_df['Order Unit Price'] = ms_quote_df['Order Unit Price'].replace('[\$]', '', regex = True).astype(float)
+    
+    # Correct issue where lead times is an integer followed by "Weeks"
+    bom_df['Mfg Std Lead Time'] = bom_df['Mfg Std Lead Time'].map(
+    lambda row: int(row.replace('Weeks', '')) if type(row) == str else int(row)
+)
+    
+    # Merges the two files on common column 'Mfr. #'
+    result = pd.merge(left = bom_df, right = ms_quote_df, on = 'Mfr. #')
+
+
+    bom_winners = pd.DataFrame()
+    ms_quote_winners = pd.DataFrame()
+
+    for index, row in result.iterrows():
+        if row['Priority'] == 'y': #
+            if row['Mfg Std Lead Time'] < row['Lead Time in Weeks']:
+                bom_winners = bom_winners.append(row)
+            elif row['Mfg Std Lead Time'] > row['Lead Time in Weeks']:
+                ms_quote_winners = ms_quote_winners.append(row)
+            elif row['Mfg Std Lead Time'] == row['Lead Time in Weeks']:
+                if row['Unit Price'] < row['Order Unit Price']:
+                    bom_winners = bom_winners.append(row)
+                elif row['Unit Price'] > row['Order Unit Price']:
+                    ms_quote_winners = ms_quote_winners.append(row)
+                else:
+                    ms_quote_winners = ms_quote_winners.append(row)
+                    bom_winners = bom_winners.append(row)
+        elif row['Priority'] == 'n':
+            if (row['Mfg Std Lead Time'] > 15 and row['Lead Time in Weeks'] > 15) or (row['Mfg Std Lead Time'] <= 15 and row['Lead Time in Weeks'] <= 15) or (row['Mfg Std Lead Time'] == row['Lead Time in Weeks']):
+                if row['Unit Price'] < row['Order Unit Price']:
+                    bom_winners = bom_winners.append(row)
+                elif row['Unit Price'] > row['Order Unit Price']:
+                    ms_quote_winners = ms_quote_winners.append(row)
+                else:
+                    ms_quote_winners = ms_quote_winners.append(row)
+                    bom_winners = bom_winners.append(row)
```

Comment:
This piece of logic is common between the bottom and top. I'm wondering if instead of repeating the code, we can make it into its own function (and name it so that its purpose is clear, such as `select_on_price`)